static/app/views/replays/detail/console/consoleMessage.tsx

```diff
@@ -45,35 +49,45 @@ function MessageFormatter({breadcrumb}: MessageFormatterProps) {
     ? sprintf.parse(message).filter(parsed => Array.isArray(parsed))
     : [];
 
-  // TODO `%c` is console specific, it applies colors to messages
-  // for now we are stripping it as this is potentially risky to implement due to xss
-  const consoleColorPlaceholderIndexes = placeholders
-    .filter(([placeholder]) => placeholder === '%c')
-    .map((_, i) => i);
-
-  // Retrieve message formatting args
-  const messageArgs = args.slice(0, placeholders.length);
-
-  // Filter out args that were for %c
-  for (const colorIndex of consoleColorPlaceholderIndexes) {
-    messageArgs.splice(colorIndex, 1);
+  if (placeholders.length) {
+    // TODO `%c` is console specific, it applies colors to messages
+    // for now we are stripping it as this is potentially risky to implement due to xss
+    const consoleColorPlaceholderIndexes = placeholders
+      .filter(([placeholder]) => placeholder === '%c')
+      .map((_, i) => i);
+
+    // Retrieve message formatting args
+    const messageArgs = args.slice(0, placeholders.length);
+
+    // Filter out args that were for %c
+    for (const colorIndex of consoleColorPlaceholderIndexes) {
+      messageArgs.splice(colorIndex, 1);
+    }
+
+    // Attempt to stringify the rest of the args
+    const restArgs = args.slice(placeholders.length).map(renderString);
+
+    const formattedMessage = isMessageString
+      ? vsprintf(message.replaceAll('%c', ''), messageArgs)
+      : renderString(message);
+
+    logMessage = [formattedMessage, ...restArgs].join(' ').trim();
+  } else {
+    // Make sure there are no objects to print in order to not get a "[Object object]"
+    const argValues: (string | number | boolean)[] = [];
+
+    for (const arg of breadcrumb.data?.arguments) {
+      if (isObject(arg) && !objectIsEmpty(arg)) {
```

Comment:
We shouldn't need to do this because `renderString` should handle pretty printing objects. If it's printing `[Object object]`, then we should fix renderString to be able to print it, instead of ignoring them.